pkg/endpointslice/controller.go

```diff
@@ -98,3 +113,14 @@ func (c *Controller) Stop() {
 
 	klog.Infof("EndpointSlice Controller stopped")
 }
+
+func (c *Controller) IsHealthy(name, namespace, clusterId string) bool {
+	key := keyFunc(name, namespace)
+	endpointInfo := c.store.Get(key)
+	if endpointInfo != nil && endpointInfo.clusterInfo != nil &&
+		endpointInfo.clusterInfo[clusterId] != nil {
+		return len(endpointInfo.clusterInfo[clusterId].ipList) > 0
+	}
```

Comment:
Slight optimization to avoid redundant map lookups:

```suggestion
	if endpointInfo != nil && endpointInfo.clusterInfo != nil {
		info := endpointInfo.clusterInfo[clusterId]
		if info != nil {
		    return len(info.ipList) > 0
		}
	}
```